===================================================================
RCS file: /cvsroot/gauche/Gauche/aclocal.m4,v
retrieving revision 1.30
retrieving revision 1.31
diff -u -r1.30 -r1.31
--- gauche/Gauche/aclocal.m4	2004/07/15 07:09:24	1.30
+++ gauche/Gauche/aclocal.m4	2004/12/21 22:49:36	1.31
@@ -93,19 +93,21 @@
 dnl   The main configure and gc's configure also use this.
 AC_DEFUN([AC_GAUCHE_OPTFLAGS],
          [
-case "$host" in
+case "$target" in
   i686-*) I686OPT="-DUSE_I686_PREFETCH";;
 esac
-if test $CC = "gcc"; then
-  case "$target" in
-    *mingw*) ;;
-    *)       GCCOPT="-fomit-frame-pointer";;
-  esac
-  case "$host" in
-   i586-*) GCCOPT="$GCCOPT -march=i586";;
-   i686-*) GCCOPT="$GCCOPT -march=i686";;
-  esac
-fi
+case "$CC" in
+  *gcc*)  # some systems may have gcc-2.95, gcc-3, etc.
+    case "$target" in
+      *mingw*) ;;
+      *)     GCCOPT="-fomit-frame-pointer";;
+    esac
+    case "$target" in
+     i586-*) GCCOPT="$GCCOPT -march=i586";;
+     i686-*) GCCOPT="$GCCOPT -march=i686";;
+    esac
+    ;;
+esac
 OPTFLAGS="$GCCOPT $I686OPT"
 AC_SUBST(OPTFLAGS)
 ])
